### ISSUE TITLE
[BUG]: Ubuntu 24.04 unable to install dependencies libicu

### ISSUE THREAD

**Issue by ZerNox** [microsoft/azure-pipelines-agent#4960]
### What happened?

In https://vstsagentpackage.azureedge.net/agent/3.243.1/vsts-agent-linux-x64-3.243.1.tar.gz

bin/installdependencies.sh  is unable to install any version libicu70

```
~$ sudo ./installdependencies.sh 
--------OS Information--------
PRETTY_NAME="Ubuntu 24.04 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
------------------------------
The current OS is Debian based
--------Debian Version--------
trixie/sid
------------------------------
/usr/bin/apt
Hit:1 http://security.ubuntu.com/ubuntu noble-security InRelease
Hit:2 https://download.docker.com/linux/ubuntu noble InRelease                                                                                        
Hit:3 http://archive.ubuntu.com/ubuntu noble InRelease                                                                                                
Hit:4 http://archive.ubuntu.com/ubuntu noble-updates InRelease                                       
Hit:5 http://archive.ubuntu.com/ubuntu noble-backports InRelease       
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
11 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libkrb5-3 is already the newest version (1.20.1-6ubuntu2.1).
libkrb5-3 set to manually installed.
zlib1g is already the newest version (1:1.3.dfsg-3.1ubuntu2.1).
zlib1g set to manually installed.
The following additional packages will be installed:
  bzip2 libdpkg-perl libfile-fcntllock-perl libfile-fnmatch-perl
Suggested packages:
  bzip2-doc bash-completion debian-keyring gcc | c-compiler binutils bzr
The following NEW packages will be installed:
  bzip2 debsums libdpkg-perl libfile-fcntllock-perl libfile-fnmatch-perl
0 upgraded, 5 newly installed, 0 to remove and 11 not upgraded.
Need to get 379 kB of archives.
After this operation, 2030 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 bzip2 amd64 1.0.8-5.1build0.1 [34.5 kB]
Get:2 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 libdpkg-perl all 1.22.6ubuntu6.1 [269 kB]
Get:3 http://archive.ubuntu.com/ubuntu noble/universe amd64 libfile-fnmatch-perl amd64 0.02-3build4 [9756 B]
Get:4 http://archive.ubuntu.com/ubuntu noble/universe amd64 debsums all 3.0.2.1 [35.5 kB]
Get:5 http://archive.ubuntu.com/ubuntu noble/main amd64 libfile-fcntllock-perl amd64 0.22-4ubuntu5 [30.7 kB]
Fetched 379 kB in 0s (1424 kB/s)                 
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package bzip2.
(Reading database ... 113352 files and directories currently installed.)
Preparing to unpack .../bzip2_1.0.8-5.1build0.1_amd64.deb ...
Unpacking bzip2 (1.0.8-5.1build0.1) ...
Selecting previously unselected package libdpkg-perl.
Preparing to unpack .../libdpkg-perl_1.22.6ubuntu6.1_all.deb ...
Unpacking libdpkg-perl (1.22.6ubuntu6.1) ...
Selecting previously unselected package libfile-fnmatch-perl.
Preparing to unpack .../libfile-fnmatch-perl_0.02-3build4_amd64.deb ...
Unpacking libfile-fnmatch-perl (0.02-3build4) ...
Selecting previously unselected package debsums.
Preparing to unpack .../debsums_3.0.2.1_all.deb ...
Unpacking debsums (3.0.2.1) ...
Selecting previously unselected package libfile-fcntllock-perl.
Preparing to unpack .../libfile-fcntllock-perl_0.22-4ubuntu5_amd64.deb ...
Unpacking libfile-fcntllock-perl (0.22-4ubuntu5) ...
Setting up libfile-fcntllock-perl (0.22-4ubuntu5) ...
Setting up bzip2 (1.0.8-5.1build0.1) ...
Setting up libdpkg-perl (1.22.6ubuntu6.1) ...
Setting up libfile-fnmatch-perl (0.02-3build4) ...
Setting up debsums (3.0.2.1) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 79.)
debconf: falling back to frontend: Readline
Scanning processes...                                                                                                                                                                                 
Scanning candidates...                                                                                                                                                                                
Scanning linux images...                                                                                                                                                                              

Running kernel seems to be up-to-date.

Restarting services...

Service restarts being deferred:
 systemctl restart unattended-upgrades.service

No containers need to be restarted.

No user sessions are running outdated binaries.

No VM guests are running outdated hypervisor (qemu) binaries on this host.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'liblttng-ust1t64' instead of 'liblttng-ust1'
The following additional packages will be installed:
  liblttng-ust-common1t64 liblttng-ust-ctl5t64
The following NEW packages will be installed:
  liblttng-ust-common1t64 liblttng-ust-ctl5t64 liblttng-ust1t64
0 upgraded, 3 newly installed, 0 to remove and 11 not upgraded.
Need to get 287 kB of archives.
After this operation, 1183 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu noble/main amd64 liblttng-ust-common1t64 amd64 2.13.7-1.1ubuntu2 [27.5 kB]
Get:2 http://archive.ubuntu.com/ubuntu noble/main amd64 liblttng-ust-ctl5t64 amd64 2.13.7-1.1ubuntu2 [74.9 kB]
Get:3 http://archive.ubuntu.com/ubuntu noble/main amd64 liblttng-ust1t64 amd64 2.13.7-1.1ubuntu2 [185 kB]
Fetched 287 kB in 1s (386 kB/s)          
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package liblttng-ust-common1t64:amd64.
(Reading database ... 113626 files and directories currently installed.)
Preparing to unpack .../liblttng-ust-common1t64_2.13.7-1.1ubuntu2_amd64.deb ...
Unpacking liblttng-ust-common1t64:amd64 (2.13.7-1.1ubuntu2) ...
Selecting previously unselected package liblttng-ust-ctl5t64:amd64.
Preparing to unpack .../liblttng-ust-ctl5t64_2.13.7-1.1ubuntu2_amd64.deb ...
Unpacking liblttng-ust-ctl5t64:amd64 (2.13.7-1.1ubuntu2) ...
Selecting previously unselected package liblttng-ust1t64:amd64.
Preparing to unpack .../liblttng-ust1t64_2.13.7-1.1ubuntu2_amd64.deb ...
Unpacking liblttng-ust1t64:amd64 (2.13.7-1.1ubuntu2) ...
Setting up liblttng-ust-common1t64:amd64 (2.13.7-1.1ubuntu2) ...
Setting up liblttng-ust-ctl5t64:amd64 (2.13.7-1.1ubuntu2) ...
Setting up liblttng-ust1t64:amd64 (2.13.7-1.1ubuntu2) ...
Processing triggers for libc-bin (2.39-0ubuntu8.3) ...
Scanning processes...                                                                                                                                                                                 
Scanning candidates...                                                                                                                                                                                
Scanning linux images...                                                                                                                                                                              

Running kernel seems to be up-to-date.

Restarting services...

Service restarts being deferred:
 systemctl restart unattended-upgrades.service

No containers need to be restarted.

No user sessions are running outdated binaries.

No VM guests are running outdated hypervisor (qemu) binaries on this host.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libssl3t64' instead of 'libssl3'
libssl3t64 is already the newest version (3.0.13-0ubuntu3.3).
libssl3t64 set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu70
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu67
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu66
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu63
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu60
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu57
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu55
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu52
'apt' failed with exit code '0'
Can't install dotnet core dependencies.
Please make sure that required repositories are connected for relevant package installer.
For issues with dependencies installation (like 'dependency was not found in repository' or 'problem retrieving the repository index file') - you can reach out to distribution owner for futher support.
You can manually install all required dependencies based on following documentation
https://docs.microsoft.com/dotnet/core/install/linux
```

### Versions

Azure devops agent 3.243.1

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Azure DevOps Server Version (if applicable)

N/A

### Operation system

Ubuntu 24.04

### Version controll system

N/A

### Relevant log output

```shell
~$ sudo ./installdependencies.sh 
--------OS Information--------
PRETTY_NAME="Ubuntu 24.04 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
------------------------------
The current OS is Debian based
--------Debian Version--------
trixie/sid
------------------------------
/usr/bin/apt
Hit:1 http://security.ubuntu.com/ubuntu noble-security InRelease
Hit:2 https://download.docker.com/linux/ubuntu noble InRelease                                                                                        
Hit:3 http://archive.ubuntu.com/ubuntu noble InRelease                                                                                                
Hit:4 http://archive.ubuntu.com/ubuntu noble-updates InRelease                                       
Hit:5 http://archive.ubuntu.com/ubuntu noble-backports InRelease       
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
11 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libkrb5-3 is already the newest version (1.20.1-6ubuntu2.1).
libkrb5-3 set to manually installed.
zlib1g is already the newest version (1:1.3.dfsg-3.1ubuntu2.1).
zlib1g set to manually installed.
The following additional packages will be installed:
  bzip2 libdpkg-perl libfile-fcntllock-perl libfile-fnmatch-perl
Suggested packages:
  bzip2-doc bash-completion debian-keyring gcc | c-compiler binutils bzr
The following NEW packages will be installed:
  bzip2 debsums libdpkg-perl libfile-fcntllock-perl libfile-fnmatch-perl
0 upgraded, 5 newly installed, 0 to remove and 11 not upgraded.
Need to get 379 kB of archives.
After this operation, 2030 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 bzip2 amd64 1.0.8-5.1build0.1 [34.5 kB]
Get:2 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 libdpkg-perl all 1.22.6ubuntu6.1 [269 kB]
Get:3 http://archive.ubuntu.com/ubuntu noble/universe amd64 libfile-fnmatch-perl amd64 0.02-3build4 [9756 B]
Get:4 http://archive.ubuntu.com/ubuntu noble/universe amd64 debsums all 3.0.2.1 [35.5 kB]
Get:5 http://archive.ubuntu.com/ubuntu noble/main amd64 libfile-fcntllock-perl amd64 0.22-4ubuntu5 [30.7 kB]
Fetched 379 kB in 0s (1424 kB/s)                 
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package bzip2.
(Reading database ... 113352 files and directories currently installed.)
Preparing to unpack .../bzip2_1.0.8-5.1build0.1_amd64.deb ...
Unpacking bzip2 (1.0.8-5.1build0.1) ...
Selecting previously unselected package libdpkg-perl.
Preparing to unpack .../libdpkg-perl_1.22.6ubuntu6.1_all.deb ...
Unpacking libdpkg-perl (1.22.6ubuntu6.1) ...
Selecting previously unselected package libfile-fnmatch-perl.
Preparing to unpack .../libfile-fnmatch-perl_0.02-3build4_amd64.deb ...
Unpacking libfile-fnmatch-perl (0.02-3build4) ...
Selecting previously unselected package debsums.
Preparing to unpack .../debsums_3.0.2.1_all.deb ...
Unpacking debsums (3.0.2.1) ...
Selecting previously unselected package libfile-fcntllock-perl.
Preparing to unpack .../libfile-fcntllock-perl_0.22-4ubuntu5_amd64.deb ...
Unpacking libfile-fcntllock-perl (0.22-4ubuntu5) ...
Setting up libfile-fcntllock-perl (0.22-4ubuntu5) ...
Setting up bzip2 (1.0.8-5.1build0.1) ...
Setting up libdpkg-perl (1.22.6ubuntu6.1) ...
Setting up libfile-fnmatch-perl (0.02-3build4) ...
Setting up debsums (3.0.2.1) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 79.)
debconf: falling back to frontend: Readline
Scanning processes...                                                                                                                                                                                 
Scanning candidates...                                                                                                                                                                                
Scanning linux images...                                                                                                                                                                              

Running kernel seems to be up-to-date.

Restarting services...

Service restarts being deferred:
 systemctl restart unattended-upgrades.service

No containers need to be restarted.

No user sessions are running outdated binaries.

No VM guests are running outdated hypervisor (qemu) binaries on this host.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'liblttng-ust1t64' instead of 'liblttng-ust1'
The following additional packages will be installed:
  liblttng-ust-common1t64 liblttng-ust-ctl5t64
The following NEW packages will be installed:
  liblttng-ust-common1t64 liblttng-ust-ctl5t64 liblttng-ust1t64
0 upgraded, 3 newly installed, 0 to remove and 11 not upgraded.
Need to get 287 kB of archives.
After this operation, 1183 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu noble/main amd64 liblttng-ust-common1t64 amd64 2.13.7-1.1ubuntu2 [27.5 kB]
Get:2 http://archive.ubuntu.com/ubuntu noble/main amd64 liblttng-ust-ctl5t64 amd64 2.13.7-1.1ubuntu2 [74.9 kB]
Get:3 http://archive.ubuntu.com/ubuntu noble/main amd64 liblttng-ust1t64 amd64 2.13.7-1.1ubuntu2 [185 kB]
Fetched 287 kB in 1s (386 kB/s)          
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package liblttng-ust-common1t64:amd64.
(Reading database ... 113626 files and directories currently installed.)
Preparing to unpack .../liblttng-ust-common1t64_2.13.7-1.1ubuntu2_amd64.deb ...
Unpacking liblttng-ust-common1t64:amd64 (2.13.7-1.1ubuntu2) ...
Selecting previously unselected package liblttng-ust-ctl5t64:amd64.
Preparing to unpack .../liblttng-ust-ctl5t64_2.13.7-1.1ubuntu2_amd64.deb ...
Unpacking liblttng-ust-ctl5t64:amd64 (2.13.7-1.1ubuntu2) ...
Selecting previously unselected package liblttng-ust1t64:amd64.
Preparing to unpack .../liblttng-ust1t64_2.13.7-1.1ubuntu2_amd64.deb ...
Unpacking liblttng-ust1t64:amd64 (2.13.7-1.1ubuntu2) ...
Setting up liblttng-ust-common1t64:amd64 (2.13.7-1.1ubuntu2) ...
Setting up liblttng-ust-ctl5t64:amd64 (2.13.7-1.1ubuntu2) ...
Setting up liblttng-ust1t64:amd64 (2.13.7-1.1ubuntu2) ...
Processing triggers for libc-bin (2.39-0ubuntu8.3) ...
Scanning processes...                                                                                                                                                                                 
Scanning candidates...                                                                                                                                                                                
Scanning linux images...                                                                                                                                                                              

Running kernel seems to be up-to-date.

Restarting services...

Service restarts being deferred:
 systemctl restart unattended-upgrades.service

No containers need to be restarted.

No user sessions are running outdated binaries.

No VM guests are running outdated hypervisor (qemu) binaries on this host.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libssl3t64' instead of 'libssl3'
libssl3t64 is already the newest version (3.0.13-0ubuntu3.3).
libssl3t64 set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu70
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu67
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu66
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu63
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu60
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu57
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu55
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package libicu52
'apt' failed with exit code '0'
Can't install dotnet core dependencies.
Please make sure that required repositories are connected for relevant package installer.
For issues with dependencies installation (like 'dependency was not found in repository' or 'problem retrieving the repository index file') - you can reach out to distribution owner for futher support.
You can manually install all required dependencies based on following documentation
https://docs.microsoft.com/dotnet/core/install/linux
```

**Comment by DenisNikulin5** [comment]
Hi @ZerNox. Thanks for reporting!

You can add libicu74 [here](https://github.com/microsoft/azure-pipelines-agent/blob/3d25a3479b5149b2bac8988615ef41616a79515a/src/Misc/layoutbin/installdependencies.sh#L102) and [here](https://github.com/microsoft/azure-pipelines-agent/blob/3d25a3479b5149b2bac8988615ef41616a79515a/src/Misc/layoutbin/installdependencies.sh#L136) to fix the problem

We will fix it in the script soon

**Comment by ZerNox** [comment]
Is 24.04 supported?

**Comment by DenisNikulin5** [comment]
Yes, it is supported

**Comment by aleksandrlevochkin** [comment]
Hi @ZerNox! The fix for this issue is available in the agent release [v3.244.1](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v3.244.1). Please, let us know if you still experience the problem with the newest version

**Comment by bdovaz** [comment]
> Yes, it is supported

The documentation is not up to date and that is why I have not updated in my case:

https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/linux-agent?view=azure-devops#check-prerequisites
